### ISSUE TITLE
Chart version 1.6.0 doesn't have its image available

### ISSUE THREAD

**Issue by mamoit** [kubernetes-sigs/nfs-ganesha-server-and-external-provisioner#115]
Since the release of version `1.6.0` a couple of hours ago, any install of the latest version of the chart is broken.
The image `k8s.gcr.io/sig-storage/nfs-provisioner:v4.0.8` is not available.

Currently, the available versions in the registry are:
``` json
[
  "v2.2.2",
  "v3.0.0",
  "v3.0.1"
]
```

Easiest workaround for us (and we should have been doing it from the start), was to pin the chart version to `1.5.0`.

The pipeline of the repo on `master` seems to be broken on the "Build nfs-ganesha base container", so it's most likely due to that.

**Comment by atalakey4work** [comment]
Guys, any update on this?

**Comment by andriisoldatenko** [comment]
i've installed prev version successfully:
```
helm install --version=1.5.0 nfs-provisioner nfs-ganesha-server-and-external-provisioner/nfs-server-provisioner -n nfs --wait
```

```
NAME: nfs-provisioner
LAST DEPLOYED: Sun Jan 22 16:06:45 2023
NAMESPACE: nfs
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
The NFS Provisioner service has now been installed.

A storage class named 'nfs' has now been created
and is available to provision dynamic volumes.

You can use this storageclass by creating a `PersistentVolumeClaim` with the
correct storageClassName attribute. For example:

    ---
    kind: PersistentVolumeClaim
    apiVersion: v1
    metadata:
      name: test-dynamic-volume-claim
    spec:
      storageClassName: "nfs"
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 100Mi
```

**Comment by atalakey4work** [comment]
Thank you @andriisoldatenko but any idea when will the latest image get published so we can upgrade to LTS?

**Comment by kvaps** [comment]
This is because of https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/pull/109

Need to find a way to fix it

**Comment by kvaps** [comment]
I have no access to approve [workflow](https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/actions/runs/3947929455)

@kmova or @msau42 could you kindly do this, please?

Also I created an issue to add admin rights for me:
https://github.com/kubernetes/org/pull/3976

**Comment by msau42** [comment]
approved

**Comment by atalakey4work** [comment]
@kvaps looking forward for your kind support to resolve the issue.

**Comment by alexandrebunn** [comment]
We've faced the same issue,tue image tag v4.0.8 isn't available.

We tried to build the image, but 'make container' ends with the error below:

```bash
Step 1/7 : ARG GANESHA_VERSION=V4.0.8
Step 2/7 : FROM ghcr.io/kubernetes-sigs/nfs-ganesha:${GANESHA_VERSION}
manifest unknown
make: *** [release-tools/build.make:105: container-nfs-provisioner] Error 1
```

Following below more info about the process:

* GO version
```bash
~/github/nfs-ganesha-server-and-external-provisioner$ go version
go version go1.16.7 linux/amd64
```

* build step
```bash
~/github/nfs-ganesha-server-and-external-provisioner$ make build
./release-tools/verify-go-version.sh "go"
mkdir -p bin
# os_arch_seen captures all of the $os-$arch seen for the current binary
# that we want to build, if we've seen an $os-$arch before it means that
# we don't need to build it again, this is done to avoid building
# the windows binary multiple times (see the default value of $BUILD_PLATFORMS)
export os_arch_seen="" && echo '' | tr ';' '\n' | while read -r os arch suffix base_image addon_image; do \
        os_arch_seen_pre=${os_arch_seen%%$os-$arch*}; \
        if ! [ ${#os_arch_seen_pre} = ${#os_arch_seen} ]; then \
                continue; \
        fi; \
        if ! (set -x; cd ./cmd/nfs-provisioner && CGO_ENABLED=0 GOOS="$os" GOARCH="$arch" go build  -a -ldflags ' -X main.version=v3.0.1-17-gced97dc -extldflags "-static"' -o "/home/alexandrebunn/github/nfs-ganesha-server-and-external-provisioner/bin/nfs-provisioner$suffix" .); then \
                echo "Building nfs-provisioner for GOOS=$os GOARCH=$arch failed, see error(s) above."; \
                exit 1; \
        fi; \
        os_arch_seen+=";$os-$arch"; \
done
+ cd ./cmd/nfs-provisioner
+ CGO_ENABLED=0
+ GOOS=
+ GOARCH=
+ go build -a -ldflags ' -X main.version=v3.0.1-17-gced97dc -extldflags "-static"' -o /home/alexandrebunn/github/nfs-ganesha-server-and-external-provisioner/bin/nfs-provisioner .
```

* make container

```bash
~/github/nfs-ganesha-server-and-external-provisioner$ make container
./release-tools/verify-go-version.sh "go"
mkdir -p bin
# os_arch_seen captures all of the $os-$arch seen for the current binary
# that we want to build, if we've seen an $os-$arch before it means that
# we don't need to build it again, this is done to avoid building
# the windows binary multiple times (see the default value of $BUILD_PLATFORMS)
export os_arch_seen="" && echo '' | tr ';' '\n' | while read -r os arch suffix base_image addon_image; do \
        os_arch_seen_pre=${os_arch_seen%%$os-$arch*}; \
        if ! [ ${#os_arch_seen_pre} = ${#os_arch_seen} ]; then \
                continue; \
        fi; \
        if ! (set -x; cd ./cmd/nfs-provisioner && CGO_ENABLED=0 GOOS="$os" GOARCH="$arch" go build  -a -ldflags ' -X main.version=v3.0.1-17-gced97dc -extldflags "-static"' -o "/home/alexandrebunn/github/nfs-ganesha-server-and-external-provisioner/bin/nfs-provisioner$suffix" .); then \
                echo "Building nfs-provisioner for GOOS=$os GOARCH=$arch failed, see error(s) above."; \
                exit 1; \
        fi; \
        os_arch_seen+=";$os-$arch"; \
done
+ cd ./cmd/nfs-provisioner
+ CGO_ENABLED=0
+ GOOS=
+ GOARCH=
+ go build -a -ldflags ' -X main.version=v3.0.1-17-gced97dc -extldflags "-static"' -o /home/alexandrebunn/github/nfs-ganesha-server-and-external-provisioner/bin/nfs-provisioner .
docker build -t nfs-provisioner:latest -f Dockerfile --label revision=v3.0.1-17-gced97dc .
Sending build context to Docker daemon  76.67MB
Step 1/7 : ARG GANESHA_VERSION=V4.0.8
Step 2/7 : FROM ghcr.io/kubernetes-sigs/nfs-ganesha:${GANESHA_VERSION}
manifest unknown
make: *** [release-tools/build.make:105: container-nfs-provisioner] Error 1
$
```

**Comment by bianchi2** [comment]
Same issue with 1.7.0

**Comment by rtrompier** [comment]
Hi guys, someone found a solution ?

**Comment by atalakey4work** [comment]
Any luck in fixing and pushing the LTS container image?

**Comment by modzilla99** [comment]
For now you could use my image: `https://github.com/modzilla99/nfs-ganesha-server-and-external-provisioner/pkgs/container/nfs-ganesha-server-and-external-provisioner` 

```
image:
  repository: ghcr.io/modzilla99/nfs-ganesha-server-and-external-provisioner
```

**Comment by blackshadev** [comment]
@kvaps the https://github.com/kubernetes/org/pull/3976 has been merged. 

Can you publish the images for us such that we can upgrade to versions after chart 1.5.0?

**Comment by atalakey4work** [comment]
Guys, any idea when this will be resolved?

**Comment by kvaps** [comment]
It seems CI-pipeline does not push actually docker image.
Sorry, I have no enough time to solve this right now.
If someone could take a look and make a fix, I'll be happy to merge it.

**Comment by revant** [comment]
I changed the container build context directory from `.` to `bin` in `release-tools/build.make` after that `make build` and `make container` worked. `nfs-provisioner:latest` was successfully built locally.

Changed `build.make`: https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/compare/master...revant:nfs-ganesha-server-and-external-provisioner:build-nfs-provisioner?expand=1

It seems only the base image (which is named ghcr.io/kubernetes-sigs/nfs-ganesha:V4.0.8) is built in `.github/workflows/docker-build.yml`.

Will it need one step to build nfs-provisioner and another step to build the final image which copies `bin/nfs-provisioner` to the base image (as specified in `deploy/docker/Dockerfile`)? What will be the final image? `k8s.gcr.io/sig-storage/nfs-provisioner` or `ghcr.io/kubernetes-sigs/nfs-ganesha`? or anything else #117?

Finally the helm chart needs to change `.Values.image` if final image is set to any name other than `k8s.gcr.io/sig-storage/nfs-provisioner`.
